controller/behavior/denormalizeable.php

```diff
@@ -0,0 +1,188 @@
+<?php
+/**
+ * @package     LOGman
+ * @copyright   Copyright (C) 2011 - 2016 Timble CVBA. (http://www.timble.net)
+ * @license     GNU GPLv3 <http://www.gnu.org/licenses/gpl.html>
+ * @link        http://www.joomlatools.com
+ */
+
+/**
+ * Denormalizeable Controller Behavior
+ *
+ * Allows for de-normalization of activities entries
+ *
+ * @author  Arunas Mazeika <https://github.com/amazeika>
+ * @package Joomlatools\Component\LOGman
+ */
+class ComActivitiesControllerBehaviorDenormalizeable extends KControllerBehaviorAbstract
+{
+    /**
+     * The resources controller
+     *
+     * @var KModelInterface
+     */
+    protected $_controller;
+
+    public function __construct(KObjectConfig $config)
+    {
+        parent::__construct($config);
+
+        $this->_controller = $config->controller;
+    }
+
+    protected function _initialize(KObjectConfig $config)
+    {
+        $config->append(array(
+            'controller' => 'resource'
+        ));
+
+        if ($this->getIdentifier()->getPackage() != 'activities')
+        {
+            $aliases = array(
+                'com:activities.controller.resource' => array(
+                    'path' => array('controller'),
+                    'name' => 'resource'
+                )
+            );
+
+            foreach ($aliases as $identifier => $alias)
+            {
+                $alias = array_merge($this->getMixer()->getIdentifier()->toArray(), $alias);
+
+                $manager = $this->getObject('manager');
+
+                // Register the alias if a class for it cannot be found.
+                if (!$manager->getClass($alias, false)) {
+                    $manager->registerAlias($identifier, $alias);
+                }
+            }
+        }
+
+        parent::_initialize($config);
+    }
+
+    protected function _afterAdd(KControllerContextInterface $context)
+    {
+        $entity = $context->result;
+
+        if ($entity->action != 'delete')
+        {
+            $context = $this->getMixer()->getContext();
+
+            $context->entity = $entity;
+
+            $this->denormalize($context);
```

Comment:
I'm but I disagree. I'm not only changing format I'm creating a whole new entity from it. Denormalize is IMO an action. The current implementation makes sense and it can work on multiple ways:

1. On after add.
2. Doing a controller->denormalize call.
3. Doing a POST _action=denormalize.

It's true though that this come with some extra overhead and that we might not need it in the end. 

Please let me know if you still want me to remove this and I'll make it so.